app/src/main/java/org/example/ChessPieces/ChessPiece.java

```diff
@@ -0,0 +1,28 @@
+package org.example.ChessPieces;
+
+import org.example.BoardPosition;
+
+// A generic class for all chess pieces.
+public abstract class ChessPiece {
+    private char mDisplayCharacter;
+    private boolean mIsWhite;
+
+    public ChessPiece(char pieceDisplayCharacter) {
+        mDisplayCharacter = pieceDisplayCharacter;
+        mIsWhite = mDisplayCharacterIsWhite(pieceDisplayCharacter);
+    }
+
+    private boolean mDisplayCharacterIsWhite(char pieceDisplayCharacter) {
+        return (pieceDisplayCharacter == '♙' || pieceDisplayCharacter == '♖' || pieceDisplayCharacter == '♘' || pieceDisplayCharacter == '♗' || pieceDisplayCharacter == '♕' || pieceDisplayCharacter == '♔');
+    }
+
+    public boolean isWhite() {
+        return mIsWhite;
+    }
+
+    public char getmDisplayCharacter() {
```

Comment:
The convention is to **not** use the `m` prefix in method names. 
Correct: `getDisplayCharacter`
Incorrect: `getmDisplayCharacter`